freshmaker/errata.py

```diff
@@ -379,3 +388,24 @@ def get_pulp_repository_ids(self, errata_id):
         data = self._errata_http_get(
             '/errata/get_pulp_packages/{}.json'.format(errata_id))
         return data.keys()
+
+    def get_cve_affected_srpm_nvrs(self, errata_id):
+        """ Get SRPM nvrs which are affected by the CVEs in errata
+
+        :param errata_id: Errata advisory ID, e.g. 25713.
+        :type errata_id: str or int
+        :return: a list of strings each of them is a source rpm nvr
+        :rtype: list
+        """
+        data = self._errata_rest_get(f"/erratum/{errata_id}/builds_by_cve")
+        srpms = []
+        for cve, data_by_product in data.items():
+            for product, product_data in data_by_product.items():
+                for build in product_data.get("builds", []):
+                    for build_nvr, build_info in build.items():
+                        # module builds are not source rpms, we need SRPMS in variants
+                        for variant, variant_data in build_info.get("variant_arch", {}).items():
+                            srpms.extend(variant_data.get("SRPMS", []))
+
+        srpm_nvrs = [x.split(".src.rpm")[0] for x in set(srpms)]
+        return srpm_nvrs
```

Comment:
Optional: I'd prefer using a set for this for efficiency and less memory usage, but I'll leave it up to you.
You can also just return a set if your calling function doesn't need a list. This tells the function caller that the order doesn't matter and that there are no duplicates, which is nice since this is always be true of this return value.
```suggestion
        srpms = set()
        for cve, data_by_product in data.items():
            for product, product_data in data_by_product.items():
                for build in product_data.get("builds", []):
                    for build_nvr, build_info in build.items():
                        # module builds are not source rpms, we need SRPMS in variants
                        for variant, variant_data in build_info.get("variant_arch", {}).items():
                            for srpm in variant_data.get("SRPMS", []):
                                srpms.add(srpm.split(".src.rpm")[0])

        return list(srpms)
```